ethereum/eth/src/test/java/tech/pegasys/pantheon/ethereum/eth/sync/fullsync/FullSyncChainDownloaderTest.java

```diff
@@ -92,15 +95,12 @@ public void setupTest() {
     syncState = new SyncState(protocolContext.getBlockchain(), ethContext.getEthPeers());
   }
 
-  // A few tests break encapsulation and access the "current task", hence casting to a concrete type
-  @SuppressWarnings("unchecked")
-  private EthTaskChainDownloader<Void> downloader(final SynchronizerConfiguration syncConfig) {
-    return (EthTaskChainDownloader<Void>)
-        FullSyncChainDownloader.create(
-            syncConfig, protocolSchedule, protocolContext, ethContext, syncState, metricsSystem);
+  private ChainDownloader downloader(final SynchronizerConfiguration syncConfig) {
+    return FullSyncChainDownloader.create(
+        syncConfig, protocolSchedule, protocolContext, ethContext, syncState, metricsSystem);
```

Comment:
Should you set up these tests to run on both versions of the downloader?

Reply:
Oh forgot this one sorry.  I typically don't bother with the aim of just switching fairly quickly and deleting the old implementation but since I went and made it so easily configurable I've parameterised the main unit test so it covers both.  Things like ATs will still only use the default but we have at least some coverage.